src/public.jl

```diff
@@ -370,3 +370,6 @@ for T in (AMDGPUDevice, AMDGPUDevice{Nothing}, CUDADevice,
     CUDADevice{Nothing}, MetalDevice, oneAPIDevice)
     @eval Adapt.adapt_storage(to::$(T), x::AbstractRange) = Adapt.adapt(to, collect(x))
 end
+
+
+isleaf(x) = Functors.isleaf(x)
```

Comment:
Add a docstring mentioning that this only overrides the behavior of types excluding `AbstractArray` / `Tuple` / `NamedTuple`